### DISCUSSION TITLE
Trap assertion error

### DISCUSSION THREAD

**Discussion by DevLeven** [NichtStudioCode/InvUI#44] [Q&A] [answered]
Hey there, I'm trying to make a player inventory claim system. that allows players to claim another players inventory after they have died. So far I've got it mostly working but I keep getting a Trap assertion error but I can't figure out what the issue is. Any ideas? I have attached a screenshot that might help.
![Screenshot 2023-07-31 210824](https://github.com/NichtStudioCode/InvUI/assets/43507866/e913fab2-c533-4223-a81d-332655afb214)

**Comment by NichtStudioCode** [comment, marked as answer]
It's this method:
```java
// net.minecraft.world.item.ItemStack
private void updateEmptyCacheFlag() {
    if (this.emptyCacheFlag && this == EMPTY) {
        throw new AssertionError("TRAP");
    } else {
        this.emptyCacheFlag = false;
        this.emptyCacheFlag = this.isEmpty();
    }
}
```

Seems like you're trying to change the amount of an empty stack (air)?

**Reply by DevLeven** [reply]
Yes this is true, if it's the inventory your talking about, it acts on claim, so in code when you click the item in the inventory the item is given to you and removed from the gui class and the item is set to null. But it mostly happens after I clear my players inventory and try and open the gui again. Any ideas on how I can fix it?
